lib/zbuff.h

```diff
@@ -45,29 +45,40 @@ extern "C" {
 *  Compiler specifics
 *****************************************************************/
 /*!
+*  ZSTD_DLL_STDCALL :
+*  Provide the ability to use stdcall linkage on public interface when building a Windows DLL
+*/
+
+#if defined(_WIN32) && defined(ZSTD_DLL_STDCALL) && (ZSTD_DLL_STDCALL==1)
+#  define ZSTDLIB_STDCALL __stdcall
+#else
+#  define ZSTDLIB_STDCALL
+#endif
+
+/*!
 *  ZSTD_DLL_EXPORT :
 *  Enable exporting of functions when building a Windows DLL
 */
+
 #if defined(_WIN32) && defined(ZSTD_DLL_EXPORT) && (ZSTD_DLL_EXPORT==1)
-#  define ZSTDLIB_API __declspec(dllexport)
+#  define ZSTDLIB_API(T) __declspec(dllexport) T ZSTDLIB_STDCALL
 #else
-#  define ZSTDLIB_API
+#  define ZSTDLIB_API(T) T ZSTDLIB_STDCALL
 #endif
 
-
 /* *************************************
 *  Streaming functions
 ***************************************/
 typedef struct ZBUFF_CCtx_s ZBUFF_CCtx;
-ZSTDLIB_API ZBUFF_CCtx* ZBUFF_createCCtx(void);
-ZSTDLIB_API size_t      ZBUFF_freeCCtx(ZBUFF_CCtx* cctx);
+ZSTDLIB_API(ZBUFF_CCtx*) ZBUFF_createCCtx(void);
```

Comment:
It's indeed pretty strange to see such construction, with the type inside parenthesis,
but if I understand correctly, `__stdcall` must be written after the function return type,
in which case, I don't see any better solution.